src/services/getExpectedApy.ts

```diff
@@ -11,16 +11,48 @@ export const getExpectedApy = (
   vt: number, // variable token balance
   margin: number, // margin until the current timestamp (margin + accrued cashflow)
   predictedApr: number, // the predicted variable APY between current and end
+  rateOracleID: number, // the rate oracle ID that suggests what type of APY is used (linear vs. compounding)
 ): [number, number] => {
-  // (rate[end]/rate[current])^(YEAR / (end-current)) - 1 = APY(current, end)
-  // rate[end]/rate[current] = (APY(current, end) + 1)^((end - current)/YEAR)
-  // rate[end]/rate[current] - 1 = (APY(current, end) + 1)^((end - current)/YEAR) - 1
-  const vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1;
+  if (end <= current) {
+    return [0, 0];
+  }
+
+  let vf = 0;
+
+  switch (rateOracleID) {
+    case 1:
```

Comment:
does this syntax imply that:

vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1; applies to cases 1-7?

Reply:
yes